File: station_ui/src/types/log.ts
/**
 * Log type definitions.
 */

/**
 * Log severity levels.
 */
export type LogLevel = 'debug' | 'info' | 'warning' | 'error';

/**
 * Log entry record.
 */
export interface LogEntry {
  /** Log ID */
  id: number;
  /** Batch ID */
  batchId: string;
  /** Execution ID (if associated with an execution) */
  executionId?: string;
  /** Log level */
  level: LogLevel;
  /** Log message */
  message: string;
  /** Timestamp */
  timestamp: Date;
}

File: station_service/models/log.py
"""
Log model definitions.
"""

from datetime import datetime
from enum import Enum
from typing import Optional

from pydantic import BaseModel


class LogLevel(str, Enum):
    """Log severity levels."""

    DEBUG = "debug"
    INFO = "info"
    WARNING = "warning"
    ERROR = "error"


class LogEntry(BaseModel):
    """Log entry record."""

    id: int
    batch_id: str
    execution_id: Optional[str] = None
    level: LogLevel
    message: str
    timestamp: datetime
